clippy_lints/src/vec.rs

```diff
@@ -20,6 +21,8 @@ use rustc_span::{sym, Span};
 pub struct UselessVec {
     pub too_large_for_stack: u64,
     pub msrv: Msrv,
+    pub span_to_lint_map: BTreeMap<Span, Option<(HirId, SuggestedType, String, Applicability)>>,
+    pub spans_sure_to_lint: BTreeSet<Span>,
```

Comment:
Is this extra set needed? It seems to me like the map on its own would be enough for determining uses

Reply:
I have removed this set, now there's only the span_to_lint_map.